src/libstd/sys/unix/os.rs

```diff
@@ -191,9 +198,15 @@ pub fn join_paths<I, T>(paths: I) -> Result<OsString, JoinPathsError>
 }
 
 impl fmt::Display for JoinPathsError {
+    #[cfg(not(target_os = "redox"))]
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         "path segment contains separator `:`".fmt(f)
     }
+
+    #[cfg(target_os = "redox")]
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        "path segment contains separator `;`".fmt(f)
```

Comment:
Could this use a constant perhaps which is defined to be either `:` or `;`?

Reply:
(same for the modifications above this as well)

Reply:
I believe so